src/utils/findPropertySource.test.ts

```diff
@@ -0,0 +1,29 @@
+import { describe, it, expect } from 'vitest'
+import { findPropertySource } from './findPropertySource'
+
+describe('findPropertySource', () => {
+  it('does return the source for objects without prototypes', () => {
+    const obj = Object.create(null)
+    obj.test = undefined
+    const source = findPropertySource(obj, 'test')
+    expect(source).toBe(obj)
+  })
+
+  it('does return the source for objects with prototypes', () => {
```

Comment:
Please drop the `does` and use `returns` instead ;) It's a nitpick, I know, but I prefer using third-person "returns"/"prevents"/etc for positive test cases and "does not X" for negative. Consistency matters.